hs_geographic_feature_resource/parse_lib.py

```diff
@@ -0,0 +1,138 @@
+# shp_metadata_dict["origin_datum"]: origin_datum
+# shp_metadata_dict["origin_unit"]: origin_unit
+
+# shp_metadata_dict["field_meta_dict"]["field_list"]: list [fieldname1, fieldname2...]
+# shp_metadata_dict["field_meta_dict"]["field_attr_dic"]: dict {"fieldname": dict{"fieldName":fieldName, "fieldTypeCode":fieldTypeCode, "fieldType":fieldType, "fieldWidth:fieldWidth, "fieldPrecision:fieldPrecision"} }
+# shp_metadata_dict["feature_count"]: feature count
+# shp_metadata_dict["geometry_type"]: geometry_type
+# shp_metadata_dict["origin_extent_dict"]: dict{"west": east, "north":north, "east":east, "south":south}
+# shp_metadata_dict["wgs84_extent_dict"]: dict{"west": east, "north":north, "east":east, "south":south}
+
+
+    shp_metadata_dict={}
+    #read shapefile
+    driver = ogr.GetDriverByName('ESRI Shapefile')
+    dataset = driver.Open(file_path)
+
+    # get layer
+    layer = dataset.GetLayer()
+    # get spatialRef from layer
+    spatialRef_from_layer = layer.GetSpatialRef()
+    print spatialRef_from_layer
+
+    if spatialRef_from_layer is not None:
+        shp_metadata_dict["origin_projection_string"] =  str(spatialRef_from_layer)
+
+        prj_name = spatialRef_from_layer.GetAttrValue('projcs')
+
+        if prj_name is None:
+            prj_name = spatialRef_from_layer.GetAttrValue('geogcs')
+        shp_metadata_dict["origin_projection_name"] = prj_name
+
+        shp_metadata_dict["origin_datum"] =  spatialRef_from_layer.GetAttrValue('datum')
+        shp_metadata_dict["origin_unit"] = spatialRef_from_layer.GetAttrValue('unit')
+    else:
+        shp_metadata_dict["origin_projection_string"] =  UNKNOWN_STR
+        shp_metadata_dict["origin_projection_name"] = UNKNOWN_STR
+        shp_metadata_dict["origin_datum"] =  UNKNOWN_STR
+        shp_metadata_dict["origin_unit"] = UNKNOWN_STR
+
+    field_list=[]
+    filed_attr_dic={}
+    field_meta_dict={"field_list":field_list, "field_attr_dict":filed_attr_dic}
+    shp_metadata_dict["field_meta_dict"]=field_meta_dict
+    #get Attributes
+    layerDefinition = layer.GetLayerDefn()
+    for i in range(layerDefinition.GetFieldCount()):
+        fieldName =  layerDefinition.GetFieldDefn(i).GetName()
+        field_list.append(fieldName)
+        attr_dict={}
+        field_meta_dict["field_attr_dict"][fieldName]=attr_dict
+
+        attr_dict["fieldName"]=fieldName
+        fieldTypeCode = layerDefinition.GetFieldDefn(i).GetType()
+        attr_dict["fieldTypeCode"]=fieldTypeCode
+        fieldType = layerDefinition.GetFieldDefn(i).GetFieldTypeName(fieldTypeCode)
+        attr_dict["fieldType"]=fieldType
+        fieldWidth = layerDefinition.GetFieldDefn(i).GetWidth()
+        attr_dict["fieldWidth"]=fieldWidth
+        fieldPrecision = layerDefinition.GetFieldDefn(i).GetPrecision()
+        attr_dict["fieldPrecision"]=fieldPrecision
+        print fieldName + " - " + fieldType+ " " + str(fieldWidth) + " " + str(fieldPrecision)
```

Comment:
@ironfroggy Sometimes I want to keep some "print" statements as they are handy for debugging and future development. Should I comment them out? or is there a better way to balance debugging needs and production performance?